### PR TITLE
FAQ, working with LLM endpoints and explaining why config-list is a list

### LINKED ISSUE #749: Why is config list a list?
Hi, it seems that the basics of the config list structure is missing or not obvious from the documentation.

Why is config list a list of models in the first place? What happens with this list of models when you pass it to an agent? Does it just use the first one or choose one with some criteria? The structure and use of the config list seems like something that should be at the very top of the documentation.

### PR DESCRIPTION
**Pull request by maxim-saplin**
Making the docs clearer about how LLM API endpoint config works.

## Related issue number

Closes #749 and #97

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [N/A] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by gagb — APPROVED** [review]
Nice clarifications. Thank you for the contributions!

File: notebook/oai_openai_utils.ipynb [hunk 1]
```diff
@@ -24,7 +24,9 @@
...
     "- `config_list_from_json`: Loads configurations from a JSON structure, either from an environment variable or a local JSON file, with the flexibility of filtering configurations based on given criteria.\n",
     "- `config_list_from_models`: Creates configurations based on a provided list of models, useful when targeting specific models without manually specifying each configuration.\n",
-    "- `config_list_from_dotenv`: Constructs a configuration list from a `.env` file, offering a consolidated way to manage multiple API configurations and keys from a single file."
+    "- `config_list_from_dotenv`: Constructs a configuration list from a `.env` file, offering a consolidated way to manage multiple API configurations and keys from a single file.\n",
+    "\n",
+    "If mutiple models are provided, the Autogen client (`OpenAIWrapper`) and agents don't choose the \"best model\" on any criteria - inference is done through the very first model and the next one is used only if the current model fails (e.g. API throtling by the provider)."
```
Comment (sonichi) [inline notebook/oai_openai_utils.ipynb:29]:
```suggestion
    "If mutiple models are provided, the Autogen client (`OpenAIWrapper`) and agents don't choose the \"best model\" on any criteria - inference is done through the very first model and the next one is used only if the current model fails (e.g. API throttling by the provider or a filter condition is unsatisfied)."
```

File: website/docs/FAQ.md [hunk 2]
```diff
@@ -20,7 +21,35 @@
...
+
+For example:
+```python
+config_list = [
+{
+"model": "gpt-4",
+"api_key": os.environ.get("AZURE_OPENAI_API_KEY"),
+"api_type": "azure",
+"base_url": os.environ.get("AZURE_OPENAI_API_BASE"),
+"api_version": "2023-03-15-preview",
```
Comment (sonichi) [inline website/docs/FAQ.md:41]:
```suggestion
"api_version": "2023-12-01-preview",
```

File: website/docs/FAQ.md [hunk 3]
```diff
@@ -47,13 +76,21 @@ You can also explicitly specify that by:
 assistant = autogen.AssistantAgent(name="assistant", llm_config={"api_key": ...})
 ```
 
+### How does an agent decide which model to pick out of the list?
+
+An agent uses the very first model available in the "config_list" and makes LLM calls against this model. If the model fail (e.g. API throtling) the agent will retry the request against the 2nd model and so on until  prompt completion is received (or throws an error if none of the models successfully completes the request). There's no implicit/hidden logic inside agents that is used to pick "the best model for the task". It is developers responsibility to pick the right models and use them with agents.
```
Comment (sonichi) [inline website/docs/FAQ.md:81]:
```suggestion
An agent uses the very first model available in the "config_list" and makes LLM calls against this model. If the model fail (e.g. API throttling) the agent will retry the request against the 2nd model and so on until  prompt completion is received (or throws an error if none of the models successfully completes the request). There's no implicit/hidden logic inside agents that is used to pick "the best model for the task". It is developers responsibility to pick the right models and use them with agents.
```

File: website/docs/FAQ.md [hunk 4]
```diff
@@ -47,13 +76,21 @@ You can also explicitly specify that by:
...
+
+Besides throtling/rotating models the 'config_list' can be useful for:
```
Comment (sonichi) [inline website/docs/FAQ.md:83]:
```suggestion
Besides throttling/rotating models the 'config_list' can be useful for:
```

File: website/docs/FAQ.md [hunk 5]
```diff
@@ -47,13 +76,21 @@ You can also explicitly specify that by:
...
+- Having a single global list of models and [filtering it](/docs/reference/oai/openai_utils#filter_config) based on certain keys (e.g. name, tag) in order to pass select models into a certain agent (e.g. use cheaper GPT 3.5 for agents solving easier tasks)
+- Using more advanced features not directly related to Agents, such as `filter_func` with [`OpenAIWrapper`](/docs/reference/oai/client#create) or [inference optimization](/docs/Examples#enhanced-inferences)
```
Comment (sonichi) [inline website/docs/FAQ.md:85]:
```suggestion
- Using more advanced features for special purposes related to inference, such as `filter_func` with [`OpenAIWrapper`](/docs/reference/oai/client#create) or [inference optimization](/docs/Examples#enhanced-inferences)
```

File: website/docs/FAQ.md (hunk 2 above)
Comment (AaronWard) [inline website/docs/FAQ.md:41]: I would format this correctly just for readability sake

File: website/docs/FAQ.md [hunk 6]
```diff
@@ -20,7 +21,35 @@
...
+"api_version": "2023-03-15-preview",
+},
+
+{
+"model": "llama-7B",
+"base_url": "http://127.0.0.1:8080",
+"api_type": "open_ai",
+}
+]
+```
```
Comment (AaronWard) [inline website/docs/FAQ.md:50]: I would format this correct just for legibility

File: website/docs/FAQ.md [hunk 7]
```diff
@@ -47,13 +76,21 @@ You can also explicitly specify that by:
 assistant = autogen.AssistantAgent(name="assistant", llm_config={"api_key": ...})
 ```
 
+### How does an agent decide which model to pick out of the list?
+
+An agent uses the very first model available in the "config_list" and makes LLM calls against this model. If the model fail (e.g. API throtling) the agent will retry the request against the 2nd model and so on until  prompt completion is received (or throws an error if none of the models successfully completes the request). There's no implicit/hidden logic inside agents that is used to pick "the best model for the task". It is developers responsibility to pick the right models and use them with agents.
+
+Besides throtling/rotating models the 'config_list' can be useful for:
+- Having a single global list of models and [filtering it](/docs/reference/oai/openai_utils#filter_config) based on certain keys (e.g. name, tag) in order to pass select models into a certain agent (e.g. use cheaper GPT 3.5 for agents solving easier tasks)
```
Comment (AaronWard) [inline website/docs/FAQ.md:84]: Screenshot 2024-01-29 at 23 06 23

Getting a 404 for this link

File: website/docs/FAQ.md [hunk 8]
```diff
@@ -47,13 +76,21 @@ You can also explicitly specify that by:
...
+- Using more advanced features not directly related to Agents, such as `filter_func` with [`OpenAIWrapper`](/docs/reference/oai/client#create) or [inference optimization](/docs/Examples#enhanced-inferences)
+
 ### Unexpected keyword argument 'base_url'
 
 In version >=1, OpenAI renamed their `api_base` parameter to `base_url`. So for older versions, use `api_base` but for newer versions use `base_url`.
 
 ### Can I use non-OpenAI models?
 
-Yes. Please check https://microsoft.github.io/autogen/blog/2023/07/14/Local-LLMs for an example.
+Yes. Autogen can work with any API endpoint which complies with OpenAI-compatible RESTful APIs - e.g. serving local LLM via FastChat or LM Studio. Please check https://microsoft.github.io/autogen/blog/2023/07/14/Local-LLMs for an example.
```
Comment (AaronWard) [inline website/docs/FAQ.md:93]: Nice clarification here 👍

**Review by AaronWard — CHANGES_REQUESTED** [review]
@maxim-saplin Looks great, nice clarifications given especially with describing how the models are chosen in sequence. I just wrote a few minor comments. When those and the typo's @sonichi mentioned i'm happy to approve 👍

**Comment by maxim-saplin** [comment]
@AaronWard , @sonichi please check the update

**Review by AaronWard — APPROVED** [review]
LGTM 💯

**Pull request merged**